apps/server/src/shared/infra/identity-management/keycloak/service/keycloak-seed.service.spec.ts

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { Test, TestingModule } from '@nestjs/testing';
+import KeycloakAdminClient from '@keycloak/keycloak-admin-client';
+import UserRepresentation from '@keycloak/keycloak-admin-client/lib/defs/userRepresentation';
+import { v1 } from 'uuid';
+import { createMock } from '@golevelup/ts-jest';
+import { Users } from '@keycloak/keycloak-admin-client/lib/resources/users';
+import { IJsonAccount, IJsonUser, IKeycloakManagementInputFiles, KeycloakManagementInputFiles } from '../interface';
+import { KeycloakAdministrationService } from './keycloak-administration.service';
+import { KeycloakSeedService } from './keycloak-seed.service';
+
+const accountsFile = 'accounts.json';
+const usersFile = 'users.json';
+let jsonAccounts: IJsonAccount[];
+let jsonUsers: IJsonUser[];
+jest.mock('node:fs/promises', () => ({
+	readFile: jest.fn().mockImplementation((path: string) => {
+		if (path.toString() === accountsFile) {
+			return Promise.resolve(JSON.stringify(jsonAccounts));
+		}
+		if (path.toString() === usersFile) {
+			return Promise.resolve(JSON.stringify(jsonUsers));
+		}
+		throw new Error();
+	}),
+}));
```

Comment:
This is a global mock and might effect all tests. It's better to do mocking stuff in the jest hooks.

Reply:
According to jest documentation this is file private

> Modules that are mocked with jest.mock are mocked only for the file that calls jest.mock. Another file that imports the module will get the original implementation even if it runs after the test file that mocks the module.

https://jestjs.io/docs/27.x/jest-object#jestenableautomock